### PR TITLE
Download custom folder

### PR DESCRIPTION
**Pull request by jeremco**
Hi

This PR adds the ability to download videos chunks in an other folder than default one. 
The main pb with previous version was that if an application changes download folder in app-settings, then all previous downloaded video will not be served.

With this PR, the download folder of a video is written in manifest.json settings file, so that it is easy to retrieve the chunks. I have added a route to express server to serve the video chunks from their donwload folder and not from default download folder.

It has to be noticed that the parameter is optional and is given in start function. If undefined, then default folder will be used.

Jérémie

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
     "jasmine": "(npm run jasmine_test || true) && npm run jasmine_report",
     "jasmine_test": "./node_modules/jasmine-node/bin/jasmine-node units/ --junitreport --output tmp/units",
     "jasmine_report": "./node_modules/junit-viewer/bin/junit-viewer --results=tmp/units --save=tmp/units_result.html",
-    "start": "electron .",
+    "start": "electron . --inspect=5858",
```
Comment (mackode) [inline package.json]: Is this change required for PR?

Comment (jeremco) [reply package.json]: You're right, it is not necessary for this PR
But to debug downstream, it is mandatory

Comment (mackode) [reply package.json]: Please remove it, rest looks nice.

**Comment by jeremco** [comment]
Please do not merge- work still in progress

**Comment by jeremco** [comment]
Hi @mackode 

Could you please check the PR we have made and then merged them ?

Thanks
